File: Courzelo-Base/src/main/java/tn/esprit/devflow/courzelo/entity/Event.java
package tn.esprit.devflow.courzelo.entity;


import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import javax.validation.constraints.Size;
import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Data
@Document
@FieldDefaults(level= AccessLevel.PRIVATE)

public class Event {

    @Id
    String idevent;
    @Size(max = 55)
    String title;
    String photo;
    int maxcapacity;
    String duration;
    LocalDate debutdate;
    Boolean price;
    @CreatedDate
    private Date createdAt;

    @LastModifiedDate
    private Date updatedAt;

    @Field("category")
    Category category;
    @DBRef
    List<EventRegistration> eventRegs;
    @DBRef
    Speaker speaker;

}

File: Courzelo_Angular22/src/app/models/Event/event.ts
import { Speaker } from "./speaker";

export class Event {


    idevent?: string;
    title?: string;
    photo? : string;
    maxcapacity?: number;
    duration?: string;
    debutdate ?: Date;
    price ?: boolean;
    category ?: Category;
    name?: Speaker;
    speaker?: Speaker;    
    photoUrl?: string;  // URL to fetch the photo
    createdAt?: Date;
    updatedAt?: Date;  
   
  }

    export enum Category{
      CONFERENCE=' CONFERENCE'
      , TEAM_BUILDING_ACTIVITIE = 'TEAM_BUILDING_ACTIVITIE', 
      COURSE_LAUNCH=' COURSE_LAUNCH', 
      VIRTUEL_TRAINING_SESSION=' VIRTUEL_TRAINING_SESSION', 
      HACKATHON=' HACKATHON', 
      VIRTUEL_RECRUITING_EVENT='VIRTUEL_RECRUITING_EVENT'
  
  
  }
